BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c

```diff
@@ -917,6 +917,13 @@ ProgramResource (
   PCI_IO_DEVICE             *PciIoDevice;
   UINT32                     Idx;
 
+  if ((BarType == PciBarTypeUnknown) || \
+      (BarType == PciBarTypeIo) || \
+      (BarType == PciBarTypeMem) || \
+      (BarType >= PciBarTypeMaxType)) {
+    return;
+  }
+

```

Comment:
why do we do this ?

Reply:
Same as previous comment

Reply:
Should we jus t do   if ((BarType == PciBarTypeUnknown) ||  ((BarType >= PciBarTypePMem64) ?